### PR TITLE
Delete post full feature

### PR DESCRIPTION
**Pull request by kaykakaraka**
created a cy task to clear the database which can be used in any test
test drove the front end delete post feature

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -63,3 +63,6 @@ typings/
 # cypress.io
 cypress/screenshots
 cypress/videos
+
+# MacOS
+.DS_Store
```
Comment (fridayshoes) [inline .gitignore:68]: Good idea, don't need those useless mac files on my PC

File: routes/posts.js [hunk 2]
```diff
@@ -6,8 +6,9 @@ const PostsController = require("../controllers/posts");
 router.get("/", PostsController.Index);
 router.post("/", PostsController.Create);
 router.get("/new", PostsController.New);
+router.post("/delete", PostsController.Delete);
```
Comment (IbrahimPala1) [inline routes/posts.js:10]: new delete route clear to see. Not sure what line 12 is doing

**Review by MalikOJBS — APPROVED** [review]

File: views/posts/index.hbs [hunk 3]
```diff
@@ -2,8 +2,9 @@
 <ul class="posts">
   {{#each posts}}
   <li>{{this.message}}
-    <form action="/posts" class="delete" method="delete">
```
Comment (IbrahimPala1) [inline views/posts/index.hbs]: Line 6 may have to go?

**Review by IbrahimPala1 — APPROVED** [review]

File: cypress/integration/user_can_delete_posts_spec.js [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
-    // submit a post
-    cy.visit("/posts");
-    cy.contains("New post").click();
-    cy.get("#new-post-form").find('[type="text"]').type("Hello, world!");
-    cy.get("#new-post-form").submit();
-    
-    // delete a post
-    cy.get('.posts').submit();
-    cy.get(".posts").should("not.contain", "Hello, world!");
+    cy.task('emptyPosts').then(() => {
```
Comment (michaelcychan) [inline cypress/integration/user_can_delete_posts_spec.js:6]: Great find for the cy.task ! Great job!

File: cypress/plugins/index.js [hunk 5]
```diff
@@ -11,7 +11,21 @@
...
+// module.exports = function() {
+//   // `on` is used to hook into various events Cypress emits
+//   // `config` is the resolved Cypress config
+// }
+
+module.exports = async (on, config) => {
+  on('task', {
+    async emptyPosts() {
+      // this is a new task that cypress understands, to delete the posts from the db
+      var con = mongoose.connect("mongodb://0.0.0.0/acebook_test");
```
Comment (michaelcychan) [inline cypress/plugins/index.js:26]: Probably the variable con might not be necessary? But as it is working, we can keep it here for now and remove it in later branches.

File: cypress/support/index.js [hunk 6]
```diff
@@ -18,3 +18,14 @@
 
 // Alternatively you can use CommonJS syntax:
 require('./commands')
+
```
Comment (michaelcychan) [inline cypress/support/index.js:21]: Line 21 - 31 can be safely removed.

File: routes/posts.js [hunk 7]
```diff
@@ -6,8 +6,9 @@ const PostsController = require("../controllers/posts");
...
 
 // implementing delete function
-router.delete("/", PostsController.Delete); // assume it's "/" first.
+// router.delete("/", PostsController.Delete); // assume it's "/" first.
```
Comment (michaelcychan) [inline routes/posts.js:12]: Line 12 can be removed as delete method cannot be used.

**Review by michaelcychan — APPROVED** [review]
Great job!

**Comment by michaelcychan** [comment]
We have reviewed the latest commit. There are some comments / old codes can be safely removed, other than that, it can be merged.

**Pull request merged**